src/realm/list.cpp

```diff
@@ -347,6 +347,37 @@ void Lst<ObjLink>::do_remove(size_t ndx)
 
 /******************************** Lst<Mixed> *********************************/
 
+Lst<Mixed>& Lst<Mixed>::operator=(const Lst<Mixed>& other)
+{
+    Base::operator=(other);
+    CollectionParent::operator=(other);
+
+    if (this != &other) {
+        // Just reset the pointer and rely on init_from_parent() being called
+        // when the accessor is actually used.
+        m_tree.reset();
+        Base::reset_content_version();
+    }
+
+    return *this;
+}
+
+Lst<Mixed>& Lst<Mixed>::operator=(Lst<Mixed>&& other) noexcept
+{
+    Base::operator=(std::move(other));
+    CollectionParent::operator=(std::move(other));
+
```

Comment:
Why isn't this
```
 Base::operator=(std::move(other));
 CollectionParent::operator=(std::move(other));
```
inside the check 
`if(this != other)`

...  if `this == other`, do we need to reassign base and parent?